apollo-router/src/plugins/headers.rs

```diff
@@ -211,10 +261,60 @@ where
     fn call(&mut self, mut req: SubgraphRequest) -> Self::Future {
         for operation in &self.operations {
             match operation {
-                Operation::Insert(config) => {
-                    req.subgraph_request
-                        .headers_mut()
-                        .insert(&config.name, config.value.clone());
+                Operation::Insert(insert_config) => {
+                    match insert_config {
+                        Insert::Static(static_insert) => {
+                            req.subgraph_request
+                                .headers_mut()
+                                .insert(&static_insert.name, static_insert.value.clone());
+                        }
+                        Insert::FromContext(insert_from_context) => {
+                            if let Some(val) = req
+                                .context
+                                .get::<_, String>(&insert_from_context.from_context)
+                                .ok()
+                                .flatten()
+                            {
+                                match HeaderValue::from_str(&val) {
+                                    Ok(header_value) => {
+                                        req.subgraph_request
+                                            .headers_mut()
+                                            .insert(&insert_from_context.name, header_value);
+                                    }
+                                    Err(err) => {
+                                        tracing::error!("cannot convert from the context into a header value: {:?}", err);
```

Comment:
it might be good to indicate in the error the subgraph name and header name (if a lot of subgraphs and headers are manipulated)